node/src/accountant/test_utils.rs

```diff
@@ -1415,18 +1436,23 @@ impl PaymentAdjuster for PaymentAdjusterMock {
         now: SystemTime,
         logger: &Logger,
     ) -> OutboundPaymentsInstructions {
-        self.adjust_payments_params
-            .lock()
-            .unwrap()
-            .push((setup, now, logger.clone()));
+        self.adjust_payments_params.lock().unwrap().push((
+            (
+                setup.adjustment,
+                setup.original_setup_msg.payload.clone(),
+                setup.original_setup_msg.agent.arbitrary_id_stamp(),
+            ),
+            now,
+            logger.clone(),
+        ));
         self.adjust_payments_results.borrow_mut().remove(0)
     }
 }
 
 impl PaymentAdjusterMock {
     pub fn is_adjustment_required_params(
         mut self,
-        params: &Arc<Mutex<Vec<(PayablePaymentsSetup, Logger)>>>,
+        params: &Arc<Mutex<Vec<((PayablePaymentsSetupMsgPayload, ArbitraryIdStamp), Logger)>>>,
```

Comment:
We talked about a way to make this less awkward; I'll leave the decision to you, because it means a minor redesign.

Reply:
My notes: The agent's trait would take a new method `dup(&self) -> Box<dyn TheTrait>` and it would be used in an explicit implementation of Clone for `PayablePaymentsSetupMsg`. Also, something like SentinelAgent could be handy for a return from the method. 

The capturing of params in this method would be affected like (PayablePaymentSetupMsg, ArbitraryIdStamp). That means we'll get a genuine message containing all the actual values except the agent (which would actually be strongly dull, empty and could respond only to the question about its ID).